api/src/backend/api/models.py

```diff
@@ -1332,3 +1335,160 @@ class Meta:
+            )
+
+        # Validate model name
+        if self.model not in self.MODEL_CHOICES:
+            raise ModelValidationError(
+                detail=f"Model must be one of: {', '.join(self.MODEL_CHOICES)}",
+                code="invalid_model",
+                pointer="/data/attributes/model",
+            )
+
+    @property
+    def api_key_decoded(self):
+        """Return the decrypted API key."""
+        if not self.api_key:
+            return None
+
+        try:
+            api_key_bytes = bytes(self.api_key)
+            decrypted_key = fernet.decrypt(api_key_bytes)
+            return decrypted_key.decode()
+
+        except InvalidToken:
+            logger.warning("Failed to decrypt API key: invalid token.")
+            return None
+        except Exception as e:
+            logger.error(f"Unexpected error while decrypting API key: {e}")
+            return None
+
+    @api_key_decoded.setter
+    def api_key_decoded(self, value):
+        """Store the encrypted API key."""
+        if not value:
+            raise ModelValidationError(
+                detail="API key is required",
+                code="invalid_api_key",
+                pointer="/data/attributes/api_key",
+            )
+
+        # Validate OpenAI API key format
+        openai_key_pattern = r"^sk-[\w-]+T3BlbkFJ[\w-]+$"
+        if not re.match(openai_key_pattern, value):
+            raise ModelValidationError(
+                detail="Invalid OpenAI API key format.",
+                code="invalid_api_key",
+                pointer="/data/attributes/api_key",
+            )
+        self.api_key = fernet.encrypt(value.encode())
+
+    def save(self, *args, **kwargs):
+        # Validate required fields
+        if not self.name:
+            raise ModelValidationError(
+                detail="Name is required",
+                code="missing_name",
+                pointer="/data/attributes/name",
+            )
+
+        if not self.model:
+            raise ModelValidationError(
+                detail="Model is required",
+                code="missing_model",
+                pointer="/data/attributes/model",
+            )
+
+        self.full_clean()
+        super().save(*args, **kwargs)
+
+    class Meta(RowLevelSecurityProtectedModel.Meta):
+        db_table = "lighthouse_config"
+
+        constraints = [
+            RowLevelSecurityConstraint(
+                field="tenant_id",
+                name="rls_on_%(class)s",
+                statements=["SELECT", "INSERT", "UPDATE", "DELETE"],
+            ),
+            # Add unique constraint for name within a tenant
+            models.UniqueConstraint(
+                fields=["tenant_id"], name="unique_lighthouse_config_per_tenant"
+            ),
+        ]
+
+        indexes = [
+            models.Index(fields=["name"], name="lighthouse_config_name_idx"),
+            models.Index(fields=["is_active"], name="lighthouse_config_active_idx"),
+        ]
```

Comment:
I don't think these are really helpful since we are going to have only one configuration per tenant and `tenant_id` is already indexed and applied through filter for every LighthouseConfiguration query.